system/inc/active_object.h

```diff
@@ -280,6 +280,10 @@ class ActiveObjectBase
         return _thread_id == std::this_thread::get_id();
     }
 
+    void setCurrentThread() {
```

Comment:
I think I'd rather have an optional parameter to start()  that indicates if threading is enabled or not, so that the thread internally takes care of setting up the thread ID. 

But really I'm not seeing why we need to reference SystemThread/ApplicationThread when threading is disabled. (I haven't read the next comment yet.)